### PR TITLE
fix: add special handling for hosted PN devices to bypass LID mapping

### PR DESCRIPTION
**Pull request by jlucaso1**
This pull request introduces significant improvements and bug fixes to the handling of LID (Login ID) and PN (Phone Number) mappings throughout the codebase, with a special focus on hosted devices. The changes enhance correctness, consistency, and reliability in JID (Jabber ID) transformations, especially for hosted device scenarios. Additionally, the PR refines device caching logic, improves message normalization, and addresses potential race conditions in E2E session injection. Comprehensive unit tests have been added to ensure the robustness of new logic.

**LID/PN Mapping and Hosted Device Handling**

* Introduced `HOSTED_DEVICE_ID` as a constant and updated logic in `LIDMappingStore` to correctly handle hosted PN and LID users, ensuring direct mapping for hosted devices and preventing unnecessary LID mapping. [[1]](diffhunk://#diff-80110f431a72a452d964c8f46b299b4069c68701ab0509780052944cfdb4592dR7) [[2]](diffhunk://#diff-35d00267183c39eafae76f69e37accfd113c5262a19de83c5930a410589f9b60R110-R116) [[3]](diffhunk://#diff-35d00267183c39eafae76f69e37accfd113c5262a19de83c5930a410589f9b60L157-R189) [[4]](diffhunk://#diff-35d00267183c39eafae76f69e37accfd113c5262a19de83c5930a410589f9b60L180-R207) [[5]](diffhunk://#diff-35d00267183c39eafae76f69e37accfd113c5262a19de83c5930a410589f9b60L204-R234) [[6]](diffhunk://#diff-cda7af2db7635420b6b14981c4d18ef344916d5e5ff07b82695ec1dfc204ed01L360-R367)
* Improved JID encoding/decoding by consistently using the new `jidEncode` utility and applying correct domain logic for hosted and standard devices throughout the mapping process. [[1]](diffhunk://#diff-35d00267183c39eafae76f69e37accfd113c5262a19de83c5930a410589f9b60L4-R13) [[2]](diffhunk://#diff-35d00267183c39eafae76f69e37accfd113c5262a19de83c5930a410589f9b60L157-R189) [[3]](diffhunk://#diff-35d00267183c39eafae76f69e37accfd113c5262a19de83c5930a410589f9b60L204-R234) [[4]](diffhunk://#diff-cda7af2db7635420b6b14981c4d18ef344916d5e5ff07b82695ec1dfc204ed01L360-R367)

**Device Cache and Message Processing**

* Refactored device cache to use the more descriptive `FullJid` type and improved device result construction with accurate server/domain resolution. [[1]](diffhunk://#diff-60bede9b89bfe9062baf0a0fa058126420659c3bbb34351fa6298c1d0c67faa1L87-R87) [[2]](diffhunk://#diff-60bede9b89bfe9062baf0a0fa058126420659c3bbb34351fa6298c1d0c67faa1L249-R272) [[3]](diffhunk://#diff-60bede9b89bfe9062baf0a0fa058126420659c3bbb34351fa6298c1d0c67faa1L319-L344)
* Enhanced message normalization logic to avoid unnecessary rewriting of hosted JIDs, preserving their format and ensuring correct handling for both remoteJid and participant fields.

**Reliability and Race Condition Fixes**

* Changed E2E session injection to process sequentially instead of using `Promise.all`, preventing race conditions in database transactions and avoiding "No session" errors during encryption. [[1]](diffhunk://#diff-66d5215ceb1308a4f1649d0c22ff4dc36a6bb5826a05ff6389314a07de4ea24fL103-R112) [[2]](diffhunk://#diff-66d5215ceb1308a4f1649d0c22ff4dc36a6bb5826a05ff6389314a07de4ea24fL130-R129)

**Testing and Minor Improvements**

* Added comprehensive unit tests for LID/PN mapping logic, including edge cases for hosted devices, direct mapping, and missing mappings.
* Minor code style and consistency improvements, such as strict equality checks and import ordering. [[1]](diffhunk://#diff-2f6a136112cda2ac0303f24b4c09666617d7926db608b4483a310c9104127825L505-R505) [[2]](diffhunk://#diff-60bede9b89bfe9062baf0a0fa058126420659c3bbb34351fa6298c1d0c67faa1L46-R48) [[3]](diffhunk://#diff-c20f2d846a8b973ca5d51a65f300f12bd449812f57b12481879c1887d471927aL8-R11) [[4]](diffhunk://#diff-033b6dd232a4d8fe514c58f7d0a6468e3872de99a834d9ccb639bf79220af665L22-L27)

These changes collectively make LID and PN mapping more robust, especially for hosted devices, and improve overall code reliability and maintainability.

### CODE REVIEW

File: src/Socket/messages-send.ts [hunk 1]
```diff
@@ -246,24 +246,30 @@ export const makeMessagesSocket = (config: SocketConfig) => {
...
-					(userDevicesCache.mget ? undefined : ((await userDevicesCache.get(user!)) as JidWithDevice[]))
+				const devices = mgetDevices?.[user] || (userDevicesCache.mget ? undefined : await userDevicesCache.get(user))
 				if (devices) {
-					const isLidJid = jid.includes('@lid')
-					const devicesWithJid = devices.map(d => ({
-						...d,
-						jid: isLidJid ? jidEncode(d.user, 'lid', d.device) : jidEncode(d.user, 's.whatsapp.net', d.device)
-					}))
+					const devicesWithJid = devices.map(deviceInfo => {
+						const finalServer = getServerFromDomainType(deviceInfo.server, deviceInfo.domainType)
```
Comment (jlucaso1) [inline src/Socket/messages-send.ts:266]: This is crucial fix to get cached devices with hosted

Comment (Salientekill) [reply src/Socket/messages-send.ts:266]: Interesting 🫡

File: src/Socket/messages-send.ts [hunk 2]
```diff
@@ -316,32 +322,15 @@ export const makeMessagesSocket = (config: SocketConfig) => {
...
-							usedLid: isLidUser
-						},
-						'Processed device with LID priority'
-					)
-				}
+				deviceResults.push({
+					...item,
+					jid: finalJid
+				})
 			}
```
[the remark this answers is not in the document]
Comment (jlucaso1) [reply src/Socket/messages-send.ts]: thanks bro love u
